appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -69,42 +82,43 @@ internal class ExperienceRenderer(
         return stateMachines.getOwner(renderContext)?.stateMachine?.state
     }
 
+    internal sealed class RenderingResult {
+        object Success : RenderingResult()
+        object CannotDisplayInterceptor : RenderingResult()
+        object CannotDisplayControlGroup : RenderingResult()
+
+        data class NoRenderContext(val experience: Experience, val renderContext: RenderContext) : RenderingResult()
+        data class StateMachineError(val experience: Experience, val error: Error) : RenderingResult()
+    }
+
     /**
      * returns true/false whether the experience is was started
      */
-    suspend fun show(experience: Experience): Boolean = with(experience) {
-        var canShow = config.interceptor?.canDisplayExperience(experience.id) ?: true
+    suspend fun show(experience: Experience): RenderingResult = with(experience) {
+        // config interceptor determines that this experience should not be displayed
+        if (config.interceptor?.canDisplayExperience(id) == false) return CannotDisplayInterceptor
```

Comment:
this change is not ok, i think - since it will now return early before checking the experiment - meaning the control group analytics are not tracked. I think the expectation is that the experiment analytics are tracked always there, like before

Reply:
you are right, I just reverted that back